massa-api/src/public.rs

```diff
@@ -381,11 +381,48 @@ impl MassaRpcServer for API<Public> {
             })
             .collect::<BTreeMap<_, _>>();
 
+        let current_cycle = last_slot
+            .unwrap_or_else(|| Slot::new(0, 0))
+            .get_cycle(api_settings.periods_per_cycle);
+
+        let cycle_duration = match api_settings.t0.checked_mul(api_settings.periods_per_cycle) {
+            Ok(cycle_duration) => cycle_duration,
+            Err(e) => return Err(ApiError::TimeError(e).into()),
+        };
+
+        let current_cycle_time_result = if current_cycle == 0 {
+            Ok(api_settings.genesis_timestamp)
+        } else {
+            let previous_cycle = current_cycle.saturating_sub(1);
```

Comment:
I still don't understand why the sub here. Regarding the example I shown : https://github.com/massalabs/massa/pull/3472#discussion_r1090358207 If we are in cycle number 2 then it's the third because we start at 0. But you multiply `cycle_duration` only one time L.400. So you will point to the timestamp at start of cycle 1. You want the one at start of cycle 2.

Can you check in sandbox what's the behaviour please

Reply:
Sorry, it was a mistake  from my first impl